### ISSUE TITLE
V6 Next.js Amplify: Ability to pass in /login or /signup for opening hosted ui

### ISSUE THREAD

**Issue by Meags27** [aws-amplify/amplify-js#12282]
### Is this related to a new or existing framework?

Next.js

### Is this related to a new or existing API?

Authentication

### Describe the feature you'd like to request

With the new v6 Amplify for Next.js, they removed the launchUri option which I used to be able to customize the urlOpener in order to change the hosted UI URL from '/oauth2/authorize?' to either '/login?' or '/signup?' so a user could click a button and get directed to either sign up or signin. 

Instead, now it calls oauthSignIn and this:

const URL = `https://${oauthConfig.domain}/oauth2/authorize?${queryString}`;
window.open(URL, SELF);

So I can't change the string "/oauth2/authorize?" easily without changing the Amplify source code. 

Thanks!

### Describe the solution you'd like

Can a parameter be passed in so we can customize the url as needed?

### Describe alternatives you've considered

N/A

### Is this something that you'd be interested in working on?

- [ ] 👋 I may be able to implement this feature request
- [ ] ⚠️ This feature might incur a breaking change

**Comment by abdallahshaban557** [comment]
Hi @Meags27  - we are trying to understand how this feature is impacting your applications and users. Would you consider this a blocker to you using v6 of the Amplify JS library?

**Comment by Meags27** [comment]
@abdallahshaban557 It's not a deal breaker to using v6, but I would vastly prefer to be able to have the option to run the site the way we did before where we have the navigation bar as "Login" and "Signup" as two different buttons to select and to go to the right page.

**Comment by abdallahshaban557** [comment]
Got it - thank you for clarifying. Would you be open to using the Amplify Authenticator connected UI component instead of hosted UI? We are planning on releasing major updates to help you fully control the UI of the default experience as well.

**Comment by Meags27** [comment]
@abdallahshaban557 I can't use the Amplify Authenticator as it doesn't support custom scopes and we wanted the option to be able to whitelist users to certain API calls in API Gateway. This issue here: https://github.com/aws-amplify/amplify-js/issues/3732

We would potentially move to using it if custom scopes were added like the others mentioned in that issue thread.

**Comment by abdallahshaban557** [comment]
Ah - got it @Meags27  - that makes sense. We are working closely with the Cognito team to resolve this issue on their end, but your reason for needing to use the HostedUI because of this limitation makes a lot of sense.

**Comment by Meags27** [comment]
I'm wanting to launch for production soon, do you have any suggestions for how I could make a workaround for this?

**Comment by subliminality** [comment]
> Hi @Meags27 - we are trying to understand how this feature is impacting your applications and users. Would you consider this a blocker to you using v6 of the Amplify JS library?

This issue is definitely preventing me from finishing the migration to V6 for one of our projects.

**Comment by israx** [comment]
@subliminality could you clarify how are you using v5 to open signup view from Hosted UI ?

**Comment by subliminality** [comment]
Yes in v5 i overrode the urlOpener on oauth when configuring Amplify
`oauth: {
  domain,
  scope,
  redirectSignIn,
  redirectSignOut,
  responseType: 'code',
  urlOpener
}`

`const urlOpener = async (url) => {
  const redirect = localStorage.getItem(LOGIN_REDIRECT) // login, signup, forgotPassword
  const signupUrl = url.replace(
    /\/oauth2\/authorize\?/,
    `/${redirect ? redirect : 'login'}?`
  )
  return launchUri(url)
}
const launchUri = (url) => {
  const windowProxy = window.open(url, SELF)
  if (windowProxy) {
    return Promise.resolve(windowProxy)
  } else {
    return Promise.reject()
  }
}`

**Comment by israx** [comment]
@subliminality  Thank you for clarifying the usage. v6 doesn't allow to modify the urlOpener. Ideally we would support this use case from the `signInWithRedirect` API itself. We will follow up with any updates on this feature request.

**Comment by mn-prp** [comment]
This upgrade broke our app until we rolled back the changes. We also override the urlOpener in v5 to go straight to sign up page on hosted ui. Like others, the hosted ui is a hard requirement for our use case. I am strongly considering just giving up on Amplify for our company and rolling our own solution, which we've already had to do on mobile, but this change is making web seem necessary too.

**Comment by trickstival** [comment]
overriding `urlOpener` is a big feature for hybrid apps that want to use an in-app browser.
I can't upgrade to amplify v6 because of that

**Comment by MatiasFacio-ParkHere** [comment]
Hi any update on this? We are badly block here and can't migrate but at the same time another issue is pushing us to migrate to v6. Kinda lost. Any help or update on this matter is very important to us and very much appreciated!

**Comment by cwomack** [comment]
Hey, @MatiasFacio-ParkHere 👋.  While we don't have any update yet tied to this issue, we understand this is potentially a migration blocking feature parity difference between v5 and v6.  Between this issue and the seemingly related #12895 and #13521, I'll ensure the topic of us removing the ability to customize how authentication URLs are opened and handled is reviewed internally.

**Comment by MatiasFacio-ParkHere** [comment]
> Hey, @MatiasFacio-ParkHere 👋. While we don't have any update yet tied to this issue, we understand this is potentially a migration blocking feature parity difference between v5 and v6. Between this issue and the seemingly related #12895 and #13521, I'll ensure the topic of us removing the ability to customize how authentication URLs are opened and handled is reviewed internally.

Thank you for your answer. I am trying to migrate now to v6 gen-1. Not easy when integrating with react-native and expo. Don't know yet how I should be handling the absense of the urlOpener property.
Currently and since 3 weeks more or less now, users are reporting that they are getting log out after an hour. This has affected the perception of our app quality, we have gotten many reviews with 'one' stars only related to this issue. 
Any update on this issue will be greatly appreciated. Thank you for taking care.

**Comment by cwomack** [comment]
@MatiasFacio-ParkHere, we hear you on the Gen 1 to Gen 2 migration story.  As for that, I can state that we are actively working on some tooling that will assist our customers with making migrations to Gen 2 easier.  This is something that is on our short term road map (and something we've discussed in the Discord server recently), so keep an eye on our docs and announcements for more on that.  

As for the urlOpener aspect tied to this issue, I'll report back as soon as there's any updates! But specifically for your "users are reporting that they are getting log out after an hour", do you know what the TTL settings are for your app's tokens? You should be able to find this within the Cognito console.
